sql/catalyst/src/main/java/org/apache/spark/sql/execution/UnsafeExternalRowSorter.java

```diff
@@ -159,6 +154,12 @@ public boolean hasNext() {
         @Override
         public UnsafeRow next() {
           try {
+            if (!alreadyCalculated) {
+              while (inputIterator.hasNext()) {
+                insertRow(inputIterator.next());
+              }
+              alreadyCalculated = true;
+            }
             sortedIterator.loadNext();
```

Comment:
`sortedIterator` is already assigned at L143. When you insert rows when first time to call `next`, can the `sortedIterator` correctly return sorted elements?

Reply:
Yes, you are right. Now I modified the `sortedIterator` after inserting rows. Due to I can only access an outer final field inside an inner class, so I used an array, is there better solution?